### ISSUE TITLE
allow redirects on declare when used as a builtin

### ISSUE THREAD

**Issue by andychu** [oilshell/oil#405]
From #400 .  This is a query rather than an assignment.

```
    declare -Ff "${1}" >/dev/null;
    ^~~~~~~
gitprompt.sh:626: Assignments shouldn't have redirects
```

**Comment by andychu** [comment]
This is because `declare -fF` is parsed as `Assignment` while `declare -f` is parsed as `SimpleCommand`

**Comment by andychu** [comment]
Fixed with the dynamic assignment change

NOTE: global assignments can still have redirects?  but maybe not ?

**Comment by andychu** [comment]
Released with 0.7.pre2: https://www.oilshell.org/release/0.7.pre2/
